File: spring_db_practice/src/main/java/com/yunhalee/spring_db_practice/config/QuerydslConfig.java
package com.yunhalee.spring_db_practice.config;

import com.yunhalee.spring_db_practice.repository.ItemRepository;
import com.yunhalee.spring_db_practice.repository.jpa.ItemQueryRepositoryV1;
import com.yunhalee.spring_db_practice.service.ItemService;
import com.yunhalee.spring_db_practice.service.ItemServiceV1;
import jakarta.persistence.EntityManager;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class QuerydslConfig {

    private final EntityManager em;

    public QuerydslConfig(EntityManager em) {
        this.em = em;
    }

    @Bean
    public ItemService itemService() {
        return new ItemServiceV1(itemRepository());
    }

    @Bean
    public ItemRepository itemRepository() {
        return new ItemQueryRepositoryV1(em);
    }

}

File: database/multidatabase/src/main/kotlin/com/yunhalee/database/multidatabase/config/QueryDslConfig.kt
package com.yunhalee.database.multidatabase.config

import com.querydsl.jpa.impl.JPAQueryFactory
import jakarta.persistence.EntityManager
import jakarta.persistence.PersistenceContext
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary

@Configuration
class QueryDslConfig {

    @PersistenceContext(unitName = "default")
    private val defaultEntityManager: EntityManager? = null

    @PersistenceContext(unitName = "a")
    private val aEntityManager: EntityManager? = null

    @Primary
    @Bean
    fun defaultQueryFactory(): JPAQueryFactory {
        return JPAQueryFactory(defaultEntityManager)
    }

    @Bean
    fun aQueryFactory(): JPAQueryFactory {
        return JPAQueryFactory(aEntityManager)
    }
}
